src/wallet/wallet_tx_builder.cpp

```diff
@@ -297,11 +305,21 @@ InvalidFundsError ReportInvalidFunds(
                : InvalidFundsReason(InsufficientFundsError(targetAmount))));
 }
 
-static void
-AddChangePayment(Payments& resolvedPayments, const ChangeAddress& changeAddr, CAmount changeAmount)
+static tl::expected<void, InputSelectionError>
+AddChangePayment(
+        const SpendableInputs& spendable,
+        Payments& resolvedPayments,
+        const ChangeAddress& changeAddr,
+        CAmount changeAmount,
+        CAmount targetAmount)
 {
     assert(changeAmount > 0);
 
+    // When spending transparent coinbase outputs, all inputs must be fully consumed.
+    if (spendable.HasTransparentCoinbase()) {
+        return tl::make_unexpected(ChangeNotAllowedError(spendable.Total(), targetAmount));
```

Comment:
Seems a little odd to only be passing in the target amount for adding it to the error, but I guess that's fine.